### PR TITLE
Update decision definition fields in the search endpoint

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

In this pull request, I am applying the latest changes in Search Decision Definition endpoint specification.

1- renaming the `DecisionDefinition` fields in the API schema (filter, sort, response) to align them with the naming used in zeebe gateway protocol. 
* `decisionId` to `dmnDecisionId`
* `name` to `dmnDecisionName`
* `decisionRequirementsId` to `dmnDecisionRequirementsId`
* `key` to `decisionKey`
* `DecisionDefinitionEntity` reflects the ES/OS field naming as it is used to fetch the documents.
* ~~I kept the ES/OS field naming in `DecisionDefinitionFilter` and `DecisionDefinitionSort`, but I feel like this is debatable. (any thoughts @ralfpuchert @nathansandi @tmetzke )?~~

2- Removal of id field as it is just a String presentation of the decisionKey (long)

3- Removal of sorting and filtering by `decisionRequirementsName` and `decisionRequirementsVersion`: these fields do not belong the decision index but to decisionRequirements index. So either we need to:
* denormalize data and save these fields in the decision index. However, this implies some extra work in the new exporter that may not be a priority for now
* do as in V1: extra query to decisionRequirements index to fetch (with scrolling) all decisionRequirement keys that match the requested `decisionRequirementsName` and `decisionRequirementsVersion`. However, this does not fit with the current search service architecture that implies one object == one index. Also sorting will not be supported with this field as in V1.
* Create a new common index that handles child-parent relationship between decision and decisionRequirements, but this will have some overhead for migration and also handling the new index in V1 endpoints.
I opened a [new issue](https://github.com/camunda/camunda/issues/20767) to handle it.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to https://github.com/camunda/camunda/issues/20654

### CODE REVIEW

File: service/src/main/java/io/camunda/service/search/filter/DecisionDefinitionFilter.java [hunk 1]
```diff
@@ -16,65 +16,50 @@
...
-    List<String> decisionRequirementsNames,
-    List<Integer> decisionRequirementsVersions,
     List<String> tenantIds)
     implements FilterBase {
 
   public static final class Builder implements ObjectBuilder<DecisionDefinitionFilter> {
 
-    private List<String> ids;
-    private List<Long> keys;
+    private List<Long> decisionKeys;
```
Comment (nathansandi) [inline service/src/main/java/io/camunda/service/search/filter/DecisionDefinitionFilter.java:30]: 💭  I believe we are using just keys over here, like previous declaration.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -1968,19 +1968,17 @@ components:
     DecisionDefinitionFilterRequest:
       type: object
       properties:
-        id:
-          type: string
-        key:
+        decisionKey:
```
Comment (nathansandi) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1971]: 🔨  We have been discussing over past definition meetings, once this is part of decision, the prefix decision before the key it is not necessary.

Does anything changed over here @tmetzke ?

**Comment by nathansandi** [comment]
> > * I kept the ES/OS field naming in `DecisionDefinitionFilter` and `DecisionDefinitionSort`, but I feel like this is debatable. (any thoughts @ralfpuchert @nathansandi @tmetzke )?
> 
> Interesting point. No strong opinion here. There will always be different names here and there between the REST API (building on the Zeebe data model) and the indices. At some point, we have to map them accordingly. I don't really mind where that happens. We just need to align on consistent handling.
> 
> One way to do it that is easy to follow (I hope): the service layer follows the REST API data model as much as possible, in the interfaces and implementation classes. We map to the ES/OS names as late as possible.

Related to this point, even if the names of ES/OS are different, I think we should follow as much as possible the Zeebe model data names (what is defined in document). And have on the Search operations similar data models to the real time.

**Review by nathansandi — CHANGES_REQUESTED** [review]
Good job :) 
Just one minor change, also some alignments that can be help us in the future over the comments.

**Comment by houssain-barouni** [comment]
> One way to do it that is easy to follow (I hope): the service layer follows the REST API data model as much as possible, in the interfaces and implementation classes. We map to the ES/OS names as late as possible.

That would not be possible for the entity `DecisionDefinitionEntity` to use the same REST API field naming. It is used to parse the documents query response. It means that we would have discrepancy between the filter (request of the service) and the entity (response of the service) which is not great. One solution is to pass by another DTO that we map from/to `DecisionDefinitionEntity` which is overkill.

**Comment by tmetzke** [comment]
> That would not be possible for the entity `DecisionDefinitionEntity` to use the same REST API field naming. It is used to parse the documents query response. It means that we would have discrepancy between the filter (request of the service) and the entity (response of the service) which is not great. One solution is to pass by another DTO that we map from/to `DecisionDefinitionEntity` which is overkill.

I see. Well, those entity classes are the latest point to map then. As I said, it's just one way. If that is the furthest "down" to the ES/OS we can go without creating additional DTOs unnecessarily, then that's it. We just need to be consistent with that then in the search endpoints. Can we work with that?

**Review by ralfpuchert — APPROVED** [review]
Besides the comments of the other reviewers LGTM.

**Comment by houssain-barouni** [comment]
> > That would not be possible for the entity `DecisionDefinitionEntity` to use the same REST API field naming. It is used to parse the documents query response. It means that we would have discrepancy between the filter (request of the service) and the entity (response of the service) which is not great. One solution is to pass by another DTO that we map from/to `DecisionDefinitionEntity` which is overkill.
> 
> I see. Well, those entity classes are the latest point to map then. As I said, it's just one way. If that is the furthest "down" to the ES/OS we can go without creating additional DTOs unnecessarily, then that's it. We just need to be consistent with that then in the search endpoints. Can we work with that?

To sum-up
`DecisionDefinitionFilter` and `DecisionDefinitionFilter` will use the REST api field naming
`DecisionDefinitionEntity` will use ES/OS field naming

Just we need to be cautious that [DecisionDefinitionServices.search](https://github.com/camunda/camunda/blob/ebfb2bc44354410525c3ae4f9f918d0d5adb16eb/service/src/main/java/io/camunda/service/DecisionDefinitionServices.java#L45-L47) would have different field namings used in the input arguments and in the returned result.
https://github.com/camunda/camunda/blob/ebfb2bc44354410525c3ae4f9f918d0d5adb16eb/service/src/main/java/io/camunda/service/DecisionDefinitionServices.java#L45-L47

**Comment by tmetzke** [comment]
> Just we need to be cautious that [DecisionDefinitionServices.search](https://github.com/camunda/camunda/blob/ebfb2bc44354410525c3ae4f9f918d0d5adb16eb/service/src/main/java/io/camunda/service/DecisionDefinitionServices.java#L45-L47) would have different field namings used in the input arguments and in the returned result.

I think that is fair and even expected. That is also what happens for Zeebe Broker calls in the services, where you put in data of one type (sometimes just plain parameters) and receive data of another type. That is what we have response mappers for, right? They map service responses (Zeebe Broker-specific, data search-layer specific) to REST responses (REST API schema).

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 2 above)
Comment (houssain-barouni) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1971]: for reference this are the renamed fields
* `decisionId` to `dmnDecisionId`
* `name` to `dmnDecisionName`
* `decisionRequirementsId` to `dmnDecisionRequirementsId`
* `key` to `decisionKey`

Comment (houssain-barouni) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1971]: @tmetzke any thoughts on this? I don't want to go against any previous decision.

File: service/src/main/java/io/camunda/service/search/filter/DecisionDefinitionFilter.java [hunk 3]
```diff
@@ -16,65 +16,50 @@
...
-    public Builder names(final List<String> values) {
+    public Builder dmnDecisionNames(final List<String> values) {
       this.names = addValuesToList(this.names, values);
       return this;
     }
 
-    public Builder names(final String value, final String... values) {
-      return names(collectValues(value, values));
+    public Builder dmnDecisionNames(final String value, final String... values) {
+      return dmnDecisionNames(collectValues(value, values));
```
[the remark this answers is not in the document]
Comment (houssain-barouni) [reply service/src/main/java/io/camunda/service/search/filter/DecisionDefinitionFilter.java]: done in https://github.com/camunda/camunda/pull/20726/commits/5fb0fac5cacf6d0b386b146fe12fa6894979bc22

**Comment by houssain-barouni** [comment]
I also added support to sorting by `decisionKey` in this commit https://github.com/camunda/camunda/pull/20726/commits/0f8f09f00c733d846a73c2338f61c31a19786079

**Review by nathansandi — APPROVED** [review]
LGTM  💯

**Pull request merged**